assets/js/RepLog/RepLogApp.js

```diff
@@ -0,0 +1,140 @@
+import React, { Component } from 'react';
+import RepLogs from './RepLogs';
+import PropTypes from 'prop-types';
+import uuid from 'uuid/v4';
+import { getRepLogs, deleteRepLog, createRepLog } from '../api/rep_log_api';
+
+export default class RepLogApp extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = Object.assign({
+            highlightedRowId: null,
+            repLogs: [],
+            numberOfHearts: 1,
+            isLoaded: false,
+            isSavingNewRepLog: false,
+            successMessage: '',
+            newRepLogValidationErrorMessage: '',
+            itemOptions: []
+        }, this.props.initialState);
```

Comment:
This opens up the possibility to inject arbitrary state and put the component into an invalid state from the start. My suggestion would be to only allow the things you need as props. The initial state would then be a combination of that what you want control over and that what you let the user inject. In order to get defaults, I'd suggest default props:

```js
RepLogApp.defaultProps = {
    repLogs: [],
    // other options
}
```

This way prop validation is easier, and there's no longer a need to merge 1 options array, you can just create one from the input because you'll know props.repLogs will always be set and will be an array.

I'd put this down as a "should change" if I'd code-review this for a company.